spec/index.bs

```diff
@@ -3622,6 +3613,19 @@ The {{GPUMapMode}} flags determine how a {{GPUBuffer}} is mapped when calling
                 1. Set |this|.{{GPUBuffer/[[pending_map]]}} to `null`,
                     and [=resolve=] |p|.
             </div>
+            <div data-timeline=content>
+                [=Content timeline=] <var data-timeline=content>map failure steps</var>:
+
+                1. If |this|.{{GPUBuffer/[[pending_map]]}} != |p|:
+
+                    Note: The map has been cancelled by {{GPUBuffer/unmap()}}.
+
+                    1. [=Assert=] |p| is already rejected.
+                    1. Return.
+                1. [=Assert=] |p| is still pending.
+                1. Set |this|.{{GPUBuffer/[[pending_map]]}} to `null`,
+                    and [=reject=] |p| with an {{OperationError}}.
```

Comment:
This means we use an OperationError in the case of the device being lost. I think this is slightly better than AbortError because right now AbortError only happens with explicit cancellation in unmap().

Reply:
Agreed. You could make a case for AbortError if the device loss was caused by an explicit destroy but I don't think anyone wants to track it that closely, nor do I think it matters to developers at that point.

Reply:
device.destroy() will actually call buffer.unmap() which is supposed to cause an AbortError (if the map and the device.destroy were on the same thread). Something to test, I'll go ahead and file this so I can mention that.